### ISSUE TITLE
[Bug] Credits are not summing up correctly for quickbooks__expenses_sales_enhanced view

### ISSUE THREAD

**Issue by jayshah-itential** [fivetran/dbt_quickbooks#91]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the issue

There are credits that are recorded in the purchase table but instead of subtracting them, they are getting counted twice in the `quickbooks__expenses_sales_enhanced` view. There is a `CREDIT` column in the `stg_quickbooks__purchase` table which is marked TRUE or FALSE and the credits should be subtracted in the final view. We noticed couple of rows in the view where credits were getting added up.

### Expected behavior

Credits should be subtracted.

### dbt Project configurations

Same configuration as defined in the package.
```
vars:
    quickbooks_database: your_destination_name
    quickbooks_schema: your_schema_name 
```

### Package versions

0.4.0 for both quickbooks and quickbooks_source

### What database are you using dbt with?

snowflake

### dbt Version

1.2

### Are you willing to open a PR to help address this issue?

- [ ] Yes.
- [ ] Yes, but I will need assistance and will schedule time during our [office hours](https://calendly.com/fivetran-solutions-team/fivetran-solutions-team-office-hours) for guidance
- [x] No.

**Comment by fivetran-joemarkiewicz** [comment]
@jayshah-itential thanks for opening this issue and raising the behavior of the purchase credits to our attention. This does seem like something we will want to account for within in our package. I also was able to confirm this understanding by diving into the [QuickBooks docs](https://developer.intuit.com/app/developer/qbo/docs/api/accounting/all-entities/purchase) and validating the behavior.

![image](https://github.com/fivetran/dbt_quickbooks/assets/74217849/2e44341a-c4b9-4cbb-927a-56e0f0e7519d)

As such I have been able to put together a starting WIP branch that incorporates the changes you are suggesting (essentially making the purchase amount a negative value when the `credits` field is `true`). If you have availability, would you be able to attempt using the branch below in place of the official dbt_quickbooks package to see if the totals are matching what you are expecting?
>**Note**: I noticed you are currently using the v0.4.0 version of the package. The WIP branch I am suggesting below is built off the latest version of the QuickBooks package. As such, I highly encourage you review the release notes of the latest versions to understand any other changes to the package.

```yml
## packages.yml

packages:
  - git: https://github.com/fivetran/dbt_quickbooks.git
    revision: bugfix/purchase-credits
    warn-unpinned: false 
```

One thing I wanted to confirm on your end, would you expect the purchase_lines to also be subtracted from the final model? In the WIP branch I am performing the -1 transformation on both the purchase `total_amount` and the purchase_line `amount` fields. Let me know your thoughts! [Relevant code changes in the WIP branch](https://github.com/fivetran/dbt_quickbooks/compare/bugfix/purchase-credits).

**Comment by jayshah-itential** [comment]
@fivetran-joemarkiewicz, I was able to test with the WIP branch and the totals seem to match now.

I did notice we had an outdated package so I did upgrade it to the latest version, hoping that would resolve the issue. Thanks for working on the fix. Let me know once this fix has been published.

**Comment by fivetran-joemarkiewicz** [comment]
That's great to hear! I will kick off the internal review process and plan to roll these updates out to the package early next week. Be sure to stay tuned here for more updates!

**Comment by fivetran-joemarkiewicz** [comment]
Hi @jayshah-itential thanks again for collaborating with us on this update to the package!

The branch with the WIP changes has since been merged and released in v0.9.1. You should see this latest version live at the top of the hour. As such, I will close this issue. Please feel free to re-open if the issue persists on your end and thank you for collaborating with our team to improve the package 😄
